app/main.py

```diff
@@ -1,2 +1,141 @@
+from typing import Any, Optional, Iterator
+from dataclasses import dataclass
+
+from app.point import Point
+
+
+@dataclass
+class Node:
+    key: Any
+    value: Any
+    node_hash: int
+
+
 class Dictionary:
-    pass
+    def __init__(self, initial_capacity: int = 8,
+                 load_factor: float = 2 / 3) -> None:
+        self.capacity: int = initial_capacity
+        self.load_factor: float = load_factor
+        self.size: int = 0
+        self.table: list[Optional[Node]] = [None] * self.capacity
+
+    def __setitem__(self, key: Any, value: Any) -> None:
+        if self.size / self.capacity >= self.load_factor:
+            self._resize()
+
+        hash_value: int = hash(key)
+        index: int = hash_value % self.capacity
+
+        while self.table[index] is not None:
+            if self.table[index].key == key:
+                # Update existing key
+                self.table[index].value = value
+                return
+            # Linear probing
+            index = (index + 1) % self.capacity
+
+        # Insert new key-value pair
+        self.table[index] = Node(key, value, hash_value)
+        self.size += 1
+
+    def __getitem__(self, key: Any) -> Any:
+        hash_value: int = hash(key)
+        index: int = hash_value % self.capacity
+
+        while self.table[index] is not None:
+            if self.table[index].key == key:
+                return self.table[index].value
+            index = (index + 1) % self.capacity
+
+        raise KeyError(f"Key '{key}' not found in dictionary.")
```

Comment:
The `__getitem__` method should raise a `KeyError` if the key is not found. You can use the `raise` keyword to raise an exception.